.github/workflows/release.yml

```diff
@@ -0,0 +1,78 @@
+name: release
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  npm-publish:
+    name: Publish to NPM & GitHub Package Registry
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: main
+      # limit releases to version changes - https://github.com/EndBug/version-check
+      - name: Check version changes
+        uses: EndBug/version-check@v1
+        id: version_check
+        with:
+          file-url: https://unpkg.com/@grafana/aws-sdk@latest/package.json
+          static-checking: localIsNew
+
+      - name: Version update detected
+        if: steps.version_check.outputs.changed == 'true'
+        run: 'echo "Version change found! New version: ${{ steps.version_check.outputs.version }} (${{ steps.version_check.outputs.type }})"'
+
+      - name: Setup .npmrc file for NPM registry
+        if: steps.version_check.outputs.changed == 'true'
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install dependencies
+        if: steps.version_check.outputs.changed == 'true'
+        run: yarn
+
+      - name: Build library
+        if: steps.version_check.outputs.changed == 'true'
+        run: yarn build
+
+      - name: Publish package to NPM
+        if: steps.version_check.outputs.changed == 'true'
+        run: npm publish --access public --scope grafana
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Setup .npmrc file for GitHub Packages
+        if: steps.version_check.outputs.changed == 'true'
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+          registry-url: 'https://npm.pkg.github.com'
+          scope: '@grafana'
+
+      - name: Publish package to Github Packages
+        if: steps.version_check.outputs.changed == 'true'
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
i don't actually see a `GITHUB_TOKEN` in this repo's settings. do we need to add that

Reply:
The GITHUB_TOKEN[ is a temporary secret](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) that is created at runtime and has certain limited permissions, so we don't need to explicitly add it. I think this should be ok without giving it any specific permissions for this action🤞🏻